File: project1/www/html/lib/seek.php
<!--author:Tang-->
<!--Last date:2019.9.2-->
<!--verison:1.0-->
<!DOCTYPE html> 
<html lang="zh-CN">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>搜索结果</title>
  <link href="../bs/css/bootstrap.css" rel="stylesheet">
  <script src="../bs/js/jquery.min.js"></script>
  <script src="../bs/js/bootstrap.js"></script>
  <script src="../bs/js/holder.js"></script>
</head>
<style>
  .demo {
    display: inline-block;
    *display: inline;
    *zoom: 1;
    overflow: hidden;
    -ms-text-overflow: ellipsis;
    text-overflow: ellipsis;
  }

  .demo:hover {
    height: auto;
    word-break: break-all;
    white-space: pre-wrap;
    text-decoration: none;
  }
  .center {
    position: fixed;
    top: 50%;
    left: 50%;
    background-color: #65;
    width: 95%;
    height: 95%;
    -webkit-transform: translateX(-50%) translateY(-50%);
  }
</style>

<body>
<script src="../bs/js/jquery.min.js"></script>
<script src="../bs/js/bootstrap.js"></script>
<script src="../bs/js/holder.js"></script>
<?php
  $host="localhost";
  $usr="lcbb3734";
  $password=3734;
  $db="lcbb3734";
  $con=mysqli_connect($servername, $username, $password, $db);
  if (!$con){
    echo '<h1>数据库连接失败，请联系管理员：XXXXXXXXXX</h1>';
  }
  //以下为查询函数
  function seek($s,$con){
    $sql = "SELECT * FROM brca_v1 where pmid REGEXP '^$s'" ;
    if(preg_match('/[^1-9]/',$s)){
      $sql = "SELECT * FROM brca_v1 where tf LIKE ".$s."'%%'"." or gene LIKE ".$s."'%s%%'";
    }
    $result = mysqli_query($con,$sql); 
    if (mysqli_num_rows($result) > 0) {
      // 输出数据
      return $result;   
      } 
    else {
      return 0;
  }
  };
    
?>



<?php   header("Content-type: text/html; charset=utf-8");
        // phpinfo();
        $s=$_POST["ser"];
        print($s);
        // 以下检查语法部分将放在前端，使用Jquery进行检查，注意改写。2019.10.14 21:26
        if (empty($s)){
          echo "<H1>you don't input anything</H1>";}
        elseif (preg_match('.[^a-zA-Z0-9].',$s)==1) {
          print("<H1>error,your input including invalidd symbol!Please input again!</H1>");
        }
        else 
          { exec("python3 /home/tan/sju3734/project1/www/html/lib/seek.py $s",$out,$return);};//待解决：改写成PHP函数 2019.10.14 22:00
        if ($out[0]=='0' or $out==null)
        {  
          echo '<h1>we don\'t find the data</h1>'; 
          echo $out;      
          }
        else{?>
        <div class="center" style="overflow: scroll" >
                <div class="table">
                  <table class="table table-hover table-bordered">
                    <thead>
                      <tr>
                        <th>PMID</th>
                        <th>TF</th>
                        <th>characteristics</th>
                        <th>gene</th>
                        <th>regulation_type</th>
                        <th>cancer_hallmark</th>
                        <th>original_text</th>
                        <th>title</th>
                      </tr>
                    </thead>                        
                    <?php $p=1;
                    foreach($out as $val){ ?>
                       <?php  if( $p==1){?>
                          <tbody><td><?php echo $val;?></td>    
                       <?php $p=$p+1;}
                        elseif ($p==8) { $p=1;?> 
                        <td><?php echo $val;?></td></tbody>
                        <?php }
                        else { ?>  
                         <td><?php echo $val;?></td>
                        <?php $p=$p+1;};?>
                    <?php }?>                 
                 </table>
                </div>
                </div> 
          <?php };
        ?>
</body>
</html>

File: project1/www/html/lib/seek.py
'''#########################################################################
# File Name: seek.py
# Author: Tang
# mail: tanzhengtang@163.com
# Created Time: 2019年09月10日 星期二 16时30分15秒
# 搜索功能   problem 目前不支持多重关键字搜索以及模糊搜索  由于转录因子和基因存在重名可能，多重显示问题
#########################################################################'''

import os,sys,MySQLdb,re
#-*- coding: utf-8 -*-
#查询函数
def checkgene(ser):
    sql="SELECT * FROM brca_v1 where tf LIKE '%s%%' or gene LIKE '%s%%'"%(str(ser),str(ser))
    try :
        int(ser)
        sql="SELECT * FROM brca_v1 where pmid REGEXP '^%s'"%str(ser)
    except:
        pass
    host="localhost"
    usr="defusr"
    password="123456"
    database="Brca"
    con=MySQLdb.connect(host,usr,password,database,use_unicode=True, charset="utf8")
    cur=con.cursor()
    # cur.execute("select COLUMN_NAME from information_schema.COLUMNS where table_name = 'brca_v1' and table_schema = 'Brca'")
    # cols=cur.fetchall()
    cur.execute(sql)
    res=cur.fetchall()
    if len(res)==0:
        print('0')
        return 0
    else:
        for i in res:
            for s in i :
                s=s.replace("α","Alpha")
                s=s.replace("β","Beta")
                s=s.replace("γ","Gamma")
                s=s.replace("δ","Delta")
                s=s.replace("ε","Epsilon")
                s=s.replace("ζ","Zeta")
                print("%s"%s)   
        # s=",".join(i)
        # p=0  #cur.execute(sql)未查询到，cur.fetchall()的元组数将会是0
        # for x in cols:
        #     if p<6 and p>0:
        #         s=s.replace("," , "    %s="%x[0],1)
        #         p=p+1
        #     elif p>=6:
        #         s=s.replace(",","\n%s="%x[0],1)
        #     else:
        #         s="pmid="+s
        #         p=p+1
    con.close()

# checkpmid(sys.argv[1])
# checkpmid(sys.argv[1])s
# checkgene(sys.argv[1])

checkgene("183")